domain/model/state/state.go

```diff
@@ -518,6 +593,90 @@ func (s *State) List(ctx context.Context) ([]coremodel.UUID, error) {
 	return models, errors.Trace(err)
 }
 
+// ListModelsForUser returns a slice of models owned or accessible by the user
+// specified by the user id. If No user or models are found an empty slice is
+// returned.
+func (s *State) ListModelsForUser(
+	ctx context.Context,
+	userID user.UUID,
+) ([]coremodel.Model, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	modelStmt := `
+SELECT md.model_uuid, md.name, cl.name, cr.name, mt.type, u.uuid, u.name, cc.name, o.name, ccn.name, l.value
```

Comment:
Isn't this the exact same sql as ListAllModels but with a WHERE clause added.
There's a lot of duplication here, also in the row scanning. Surely we can have a lower case listModels() which takes an optional user to filter on as a parameter. If the user is not empty, add the where clause. Everything else stays the same.

Reply:
Yeah we can, happy to it just seems just as trivial to leave it as it's own entity. This SQL will get cleaned up with the DDL changes.

Repeating ourselves is not always a bad thing.

Reply:
Fair enough, the clean up can fix it. And some repeating is fine. Bit IMO not for non-trivial SQL where is super trivial to get the reuse.